### ISSUE TITLE
Could not find any implementation for node failure of TensorRT 8.6.13 when running MHA on GPU A10

### ISSUE THREAD

**Issue by XuDeshengCat** [NVIDIA/TensorRT#4189]
## Environment

**TensorRT Version**:8.6.13

**NVIDIA GPU**:A10

**NVIDIA Driver Version**:

**CUDA Version**:12.3

**CUDNN Version**:

Operating System:

Python Version (if applicable):

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files

**Model link**:

## Steps To Reproduce

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by lix19937** [comment]
ref https://github.com/NVIDIA/TensorRT/issues/4167#issuecomment-2388490043

**Comment by yuanyao-nv** [comment]
Also, suggest that you try the latest TRT version.

**Comment by XuDeshengCat** [comment]
> Also, suggest that you try the latest TRT version.

Thanks for your reply.
However, due to the requirements of the project, I had to use version 8.6.13 of TensorRT. Is there any other way I can optimize MHA? Like some open source plugin library?

**Comment by XuDeshengCat** [comment]
> ref [#4167 (comment)](https://github.com/NVIDIA/TensorRT/issues/4167#issuecomment-2388490043)

Thanks for your reply.
I noticed that this issue is about version 10.0 of TensorRT. I looked at the [document](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-861/developer-guide/index.html#enable-fusion) for version 8.6.1 and there was no mention of MHA or attention. Are MHA optimization rules always the same?

**Comment by lix19937** [comment]
@XuDeshengCat  Yes, 8.6.1 not support any mha fusion, there is a custom plugin https://github.com/NVIDIA/TensorRT/tree/release/8.6/plugin/bertQKVToContextPlugin, which accelerate `softmax(K'Q/sqrt(HeadSize)) *v ` ops,  maybe help.

**Comment by XuDeshengCat** [comment]
> [@XuDeshengCat](https://github.com/XuDeshengCat) Yes, 8.6.1 not support any mha fusion, there is a custom plugin https://github.com/NVIDIA/TensorRT/tree/release/8.6/plugin/bertQKVToContextPlugin, which accelerate `softmax(K'Q/sqrt(HeadSize)) *v ` ops, maybe help.

Thanks.
I cloned the Tensorrt project, compiled the whole project, and found that the` bertQKVToContextPlugin `was compiled to` libnvinfer_plugin.so`. If I want to use this plugin, do I just need to add the command in trtexec:` --plugins=/cudalib/third-part/tensorrt/x64-linux/lib/libnvinfer_plugin.so \`, and meet the input shape requirements of this plugin, tensorrt will try to use this plugin in the process of building the engine?

**Comment by lix19937** [comment]
Yes, bertQKVToContextPlugin was build in libnvinfer_plugin.so, this plugin also has adaptability conditions, so you need check the conditions.  If your case match conditions, use trtexec and not add `--plugins=spec`.   
If your case not match conditions, you need modify the plugin impl, and then recompile the plugin src.
